### PR TITLE
DataGrid.Pdf - Set backgroundColor for header, group, total rows

### PR DESCRIPTION
**Pull request by KuznetsovVN**

### CODE REVIEW

**Review by IgnatovDan — APPROVED** [review]

File: js/exporter/jspdf/export_data_grid_2.js [hunk 1]
```diff
@@ -25,13 +25,15 @@ function exportDataGrid(doc, dataGrid, options) {
 
             pdfGrid.startNewTable(options.drawTableBorder, options.topLeft);
 
+            const { headerStyles, groupStyles, totalStyles } = options;
+            const rowStyles = { headerStyles, groupStyles, totalStyles };
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js]: Destructuring and immediate constructing back looks strange.
How the same scenarios are managed in our asp/mvc/winforms components?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
+import { exportDataGrid } from 'exporter/jspdf/export_data_grid_2';
+
+const JSPdfStylesTests = {
+    runTests(moduleConfig, createMockPdfDoc, createDataGrid) {
+
+        const onRowExporting = (e) => { e.rowHeight = 16; };
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js]: Will we introduce an argument for 'one height for all rows' scenario?

File: testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+                        { dataField: 'f2' },
+                    ],
+                    dataSource: [
+                        { f1: 'f1_1', f2: 'f1_2' },
+                    ],
+                });
+
+                const expectedLog = [
+                    'setFillColor,#808080', 'rect,10,15,90,16,F',
+                    'text,F1,10,23,{baseline:middle}', 'setLineWidth,1', 'rect,10,15,90,16',
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js]: We can test rect in one test and remove rects for all the others to reduce expected logs, what do you think?
Or, can we draw both the rect and fill in one 'rect' call?

File: testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+                    'setFillColor,#808080', 'rect,100,15,80,16,F',
+                    'text,F2,100,23,{baseline:middle}', 'setLineWidth,1', 'rect,100,15,80,16',
+                    'text,f1_1,10,39,{baseline:middle}', 'setLineWidth,1', 'rect,10,31,90,16',
+                    'text,f1_2,100,39,{baseline:middle}', 'setLineWidth,1', 'rect,100,31,80,16'
+                ];
+
+                exportDataGrid(doc, dataGrid, { topLeft: { x: 10, y: 15 }, columnWidths: [ 90, 80 ], onRowExporting, ...rowTypeStyles }).then(() => {
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/jspdfParts/jspdf_v2.dataGrid.styles.tests.js]: Styles were constructed in an object and destructed here.
Internally, styles will be constructed back to a single object.
Can we pass an object without destruct/construct?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

**Review by IgnatovDan — APPROVED** [review]

**Review by novsstation — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**